src/projection.jl

```diff
@@ -310,6 +322,45 @@ function (project::ProjectTo{<:Tangent{<:Tuple}})(dx::Tuple)
     dy = map((f, x) -> f(x), project.elements, dx)
     return project_type(project)(dy...)
 end
+function (project::ProjectTo{<:Tangent{<:NamedTuple}})(dx::NamedTuple)
+    dy = _project_namedtuple(backing(project), dx)
+    return project_type(project)(; dy...)
+end
+
+# Diffractor returns not necessarily a named tuple with all keys and of the same order as
```

Comment:
what does Diffractor have to do with anything, and why does it return a namedtuple?
It should be a `Tangent`.

Reply:
It refers to https://github.com/JuliaDiff/ChainRulesCore.jl/pull/515#issuecomment-965855697. The `Tangent`s are already unpacked at this stage.